internal/common/server.go

```
@@ -0,0 +1,141 @@
+// AuthMiddleware creates an HTTP middleware for authentication
+func AuthMiddleware(provider auth.Provider, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Extract authentication information
+		authInfo, err := provider.Authenticate(r)
+		if err != nil {
+			ReturnJSONError(w, http.StatusUnauthorized, "Unauthorized")
+			return
+		}
+
+		// Store authentication info in the request context
+		ctx := context.WithValue(r.Context(), "auth_info", authInfo)
```

golangci-lint (1.64.8)

SA1029: should not use built-in type string as key for value; define your own type to avoid collisions

(staticcheck)